src/clients/api/mutations/useClaimVaiVaultReward.ts

```diff
@@ -0,0 +1,41 @@
+import { MutationObserverOptions, useMutation } from 'react-query';
+
+import {
+  queryClient,
+  claimVaiVaultReward,
+  IClaimVaiVaultRewardInput,
+  ClaimVaiVaultRewardOutput,
+} from 'clients/api';
+import FunctionKey from 'constants/functionKey';
+import { useVaiVaultContract } from 'clients/contracts/hooks';
+
+type Options = MutationObserverOptions<
+  ClaimVaiVaultRewardOutput,
+  Error,
+  Omit<IClaimVaiVaultRewardInput, 'vaiVaultContract'>
+>;
+
+const useClaimVaiVaultReward = (options?: Options) => {
+  const vaiVaultContract = useVaiVaultContract();
```

Comment:
Should we be using the ProxyContract here? 

We are checking for failure events but this contract doesn't send failure events. The Proxy contract does

Reply:
I don't think so, the current implementation doesn't use the proxy.

Reply:
In that case we don't have to check for the Error event. The VAiVaultContract doesn't send the `fail` or `failOpaque` events. It doesn't hurt anything though